lib/internal/process/promises.js

```diff
@@ -1,15 +1,44 @@
 'use strict';
 
+const { ERR_INVALID_ENV_VALUE } = require('internal/errors').codes;
+
 const { safeToString } = process.binding('util');
 
 const maybeUnhandledPromises = new WeakMap();
-const pendingUnhandledRejections = [];
-const asyncHandledRejections = [];
+let pendingUnhandledRejections = [];
+let asyncHandledRejections = [];
+const kFromPromise = true;
 let lastPromiseId = 0;
 
+const kSilent = 0;
+const kWarn = 1;
+const kErrorOnGC = 2;
+const kError = 3;
+
 exports.setup = setupPromises;
 
+function unhandledState() {
+  let env = process.env.NODE_UNHANDLED_REJECTION;
+  if (env) {
+    env = env.toUpperCase();
+    if (env === 'ERROR') {
+      return kError;
+    }
+    if (env === 'SILENT') {
+      return kSilent;
+    }
+    if (env === 'ERROR_ON_GC') {
+      return kErrorOnGC;
+    }
+    if (env !== 'WARN') {
```

Comment:
nit: check for `=== 'WARN'` and return `kWarn` and put the `ERR_INVALID_END_VALUE` last for readability

Reply:
That would be a changed behavior. Right now it also works for the case that the environment variable is not set. In that case it works like `WARN`.

Reply:
Got it, let's revisit this when doing the v11 change to abort by default